src/utils.cairo

```diff
@@ -194,14 +194,30 @@ pub fn byte_array_to_felt252(byte_array: @ByteArray) -> felt252 {
     let byte_shift = 256;
     let mut value = 0;
     let mut i = 0;
-    let byte_array_len = byte_array.len();
+    let byte_array_len = byte_array.len().into();
     while i < byte_array_len {
-        value = value * byte_shift + byte_array[i].into();
+        value += byte_array[i].into() * power(byte_shift, i.into()).into();
```

Comment:
Doing power operations is super expensive, it might be better to just change the orderer of the loop to start at `byte_array_len` and decrease to 0. Like `let mut i = `byte_array_len` and `while i != 0` with `i -= 1`.

Reply:
doing this gives me an error on execution, let me see if I can find another way to do this